res_syntax/tests/ppx/react/expected/defaultValueProp.res.txt

```diff
@@ -0,0 +1,40 @@
+module C0 = {
+  type props<'a, 'b> = {a?: 'a, b?: 'b}
+  @react.component
+  let make = (props: props<'a, 'b>) => {
+    let b = switch props.b {
+    | Some(b) => b
+    | None => a * 2
```

Comment:
If you copy paste this, you'll get a `a not found`.
The oder should be: first `a` then `b`.
Also, it should not be `let ... and` but `let ... let` or it won't compile.

Reply:
Actually, `a` should refer to the outside scope of make function. The original code is this:
```rescript
@react.component
let make = (~a=2, ~b=a*2) => <div />
```
So, `a` in `(a* 2)` is referring to out of this make function.

Reply:
This is quite tricky than I guess. It has to be affected by the order, but it should not.
```rescript
let make = (props) => {
  let a = 2 // 1
  let b = a * 2 // <- a should be reference to the outside of make function, but it refers to 1
```

Reply:
We transform the optional labelled argument to nolabel argument `props`. So there no where to handle the default value that is the root cause of this issue, I guess.